### DISCUSSION TITLE
Using other port number than 443 for https does not work

### DISCUSSION THREAD

**Discussion by helander** [pocketbase/pocketbase#3041] [Q&A] [answered]
I have tried - as a non-root user - to start the pocketbase server using —https with a port number other than 443. Everything seems to start, but I can not connect to the server via the specified port. Any idea what the problem might be?

**Comment by ganigeorgiev** [comment]
PocketBase uses Let's encrypt for the TLS certificate and the validation can happen (as far as I know) only on 80 or 443 port (or via explicitly defined DNS record for the acme 01 challenge).

To help you with the debugging, you can start PocketBase with `--debug` and see if there are any messages printed in the console (or in your dedicated stderr output).

If you are trying to host multiple applications on a single machine, my recommendation would be to use a reverse proxy (nginx, apache, caddy, etc.) that will listen on 80/443 and will proxy the requests to your other applications that could listen to some other local port. An example nginx config you could find in https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

**Reply by ganigeorgiev** [reply, marked as answer]
_You can also check https://letsencrypt.org/docs/challenge-types._

**Reply by helander** [reply]
Thanks for the answer. The link to letsencrypt makes it very clear. I used to have a reverse proxy (traefik) on the host inquestion. I will have to redeploy it.

**Comment by LyraHarring** [comment]
I ran into the same thing. PocketBase with https works only if it can find a valid cert/key and the port is actually open. By default, HTTPS usually expects [port 443](https://certera.com/blog/what-is-port-443-detailed-guide-on-https-port-443/), but you can use another port if nothing blocks it. In my case, the issue was the firewall or permissions on that custom port. Try checking if the cert paths are correct and if your port is allowed through the firewall. Hope it helps!
